exo-kudos-services/src/main/java/org/exoplatform/addon/kudos/listener/IntegrationGamificationListener.java

```diff
@@ -0,0 +1,54 @@
+package org.exoplatform.addon.kudos.listener;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.exoplatform.addon.kudos.model.Kudos;
+import org.exoplatform.addon.kudos.service.KudosService;
+import org.exoplatform.services.listener.Event;
+import org.exoplatform.services.listener.Listener;
+import org.exoplatform.services.listener.ListenerService;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+
+/**
+ * A listener to add comment or activity
+ */
+public class IntegrationGamificationListener extends Listener<KudosService, Kudos> {
+  private static final Log LOG = ExoLogger.getLogger(IntegrationGamificationListener.class);
+  
+  private ListenerService  listenerService;
+
+  public IntegrationGamificationListener(ListenerService  listenerService) {
+    this.listenerService = listenerService;
+  }
+
+  @Override
+  public void onEvent(Event<KudosService, Kudos> event) throws Exception {
+    Kudos kudos = event.getData();
+ 
+    try {
+      Map<String, String> gam = new HashMap<>();
+      gam.put("ruleTitle", "sendKudos");
+      gam.put("senderId", kudos.getSenderId());
+      gam.put("receiverId", kudos.getSenderId());
+      gam.put("object", "");
+      listenerService.broadcast("exo.gamification.generic.action", gam, "");
+  } catch (Exception e) {
+    LOG.error("Cannot broadcast gamification event");
+  }
+    try {
+      Map<String, String> gam = new HashMap<>();
+      gam.put("ruleTitle", "receiveKudos");
+      gam.put("senderId", kudos.getReceiverId());
```

Comment:
I think you should put the kudos sender here:
gam.put("senderId", kudos.getSenderId());

Ps in gamification the sender is user who do the action (here the one who send the kudo) and the receiver id the user who will get the points (here the one who receive the kudo)

And please be sure thet the senderId and receiverId in the gamification object should be the userName not the social identity ID

Reply:
done